### PR TITLE
[CAT-1032 + more] Dependency integration and update of some libraries.

### PR DESCRIPTION
**Pull request by obusher**
* mockito-core (includes dexmaker )
* xstream dependency integration
* libgdx update from 1.5.5 to 1.6.2
* Delete of local unused libraries

### CODE REVIEW

File: catroidSourceTest/build.gradle [hunk 1]
```diff
@@ -5,7 +5,6 @@ repositories{
 apply plugin: 'java'
 
 dependencies {
-    testCompile files( project(":").buildDir.getPath() + '/intermediates/classes/debug' )
```
Comment (obusher) [inline catroidSourceTest/build.gradle]: @aried3r , @xorrizon: Why do we need to compile the `/intermediates/classes/debug`directory?

**Comment by obusher** [comment]
ok to test

File: catroidSourceTest/build.gradle (hunk 1 above)
Comment (xorrizon) [inline catroidSourceTest/build.gradle]: Some tests need access to the compiled classes instead only on sources. But to find them they need to be declared as dependency.

If those tests have been removed this line is not needed anymore.

File: build.gradle [hunk 2]
```diff
@@ -54,8 +54,9 @@ dependencies {
     compile 'com.google.guava:guava:18.0'
     compile 'com.squareup.okhttp:okhttp:2.1.0-RC1'
     compile 'com.squareup.okhttp:okhttp-urlconnection:2.1.0-RC1'
-    compile 'com.github.kevinsawicki:http-request:5.6'
+    compile 'com.github.kevinsawicki:http-request:6.0'
```
Comment (aried3r) [inline build.gradle]: When will the release branch be merged into the develop branch? http-request is not used anymore in the release branch afaik.

**Comment by aried3r** [comment]
> Delete of local unused libraries

What about `/libraryProjects/actionbarsherlock`?

Also, [libgdx 1.5.6 has been released](www.badlogicgames.com/wordpress/?p=3666), in case you want to update that, too.

**Comment by aried3r** [comment]
Please resolve the merge conflicts.

**Comment by obusher** [comment]
ok to test

**Comment by obusher** [comment]
ok to test

**Comment by avihric** [comment]
Is it necessary to have 4 commits for this rather small changes? You should also stick to the commit message rules e.g: `CAT-1032 Some meaningful msg`

**Comment by robertpainsi** [comment]
Since this PR updates a bunch of libraries, maybe you can smuggle https://github.com/Catrobat/Catroid/pull/1249 in?

**Comment by obusher** [comment]
@avihric Will be done if the jenkins run is green. (Was already green pre-merge-stop)
I am not 100% sure if some of the library updates causes some test-fails or new timing issues got introduced...

**Comment by obusher** [comment]
ok to test

**Comment by obusher** [comment]
ok to test

**Comment by tobijdc** [comment]
Checkstyle 6.8 released =D
https://github.com/checkstyle/checkstyle/releases/tag/checkstyle-6.8

Also PMD version is out of date (maybe something for the next ticket)

**Comment by obusher** [comment]
Checkstyle update should be done in #1249 .
As far as i remember there exists already a jirra issue concerning the PMD update.

**Pull request merged**